### PR TITLE
Fix e2e tests for CRT-1401

### PR DESCRIPTION
**Pull request by sbryzak**
Updated e2e tests for https://github.com/codeready-toolchain/registration-service/pull/254

### CODE REVIEW

**Review by alexeykazakov — APPROVED** [review]

File: testsupport/wait/host.go [hunk 1]
```diff
@@ -578,6 +578,33 @@ func (a *HostAwaitility) WaitForUserSignup(name string, criteria ...UserSignupWa
...
+	a.T.Logf("waiting for UserSignup '%s' or '%s' in namespace '%s' to match criteria", userID, username, a.Namespace)
+	var userSignup *toolchainv1alpha1.UserSignup
+	err := wait.Poll(a.RetryInterval, a.Timeout, func() (done bool, err error) {
+		obj := &toolchainv1alpha1.UserSignup{}
+		if err := a.Client.Get(context.TODO(), types.NamespacedName{Namespace: a.Namespace, Name: userID}, obj); err != nil {
+			if errors.IsNotFound(err) {
+				if err := a.Client.Get(context.TODO(), types.NamespacedName{Namespace: a.Namespace, Name: username}, obj); err != nil {
+					if errors.IsNotFound(err) {
+						return false, nil
+					}
```
Comment (MatousJobanek) [inline testsupport/wait/host.go:592]: if there is an error and is not "NotFound" then we should return the error
```suggestion
					}
					return false, err
```

Comment (sbryzak) [reply testsupport/wait/host.go:592]: Fixed

**Review by alexeykazakov — COMMENTED** [review]
Seems to be some actual problem with the e2e tests or maybe with the paired reg-service PR which should be fixed:

```
   member.go:118: failed to find UserAccount with matching criteria:
        ----
        actual:
        metadata:
          annotations:
            toolchain.dev.openshift.com/user-email: testuser-406a621c-5634-4dc5-a84d-96b833b3cfe8@test.com
          creationTimestamp: "2022-03-18T23:46:49Z"
          finalizers:
          - finalizer.toolchain.dev.openshift.com
          generation: 1
          labels:
            toolchain.dev.openshift.com/tier: cheesecake
          name: cheesecakelover07
          namespace: toolchain-member-18233743
          resourceVersion: "50352"
          uid: 1bd1af6d-472e-45a5-be9c-6f978199c6fb
        spec:
          nsLimit: ""
          userID: c115b698-0721-4b8c-8413-a1c29c4cba0c
        status:
          conditions:
          - lastTransitionTime: "2022-03-18T23:46:49Z"
            reason: Provisioned
            status: "True"
            type: Ready
        
        ----
        diffs:
        expected specs to match:   interface{}(
        - 	&v1alpha1.UserAccountSpec{
        - 		UserID:              "cheesecakelover07",
        - 		UserAccountSpecBase: v1alpha1.UserAccountSpecBase{NSTemplateSet: &v1alpha1.NSTemplateSetSpec{}},
        - 	},
        + 	v1alpha1.UserAccountSpec{
        + 		UserID:              "c115b698-0721-4b8c-8413-a1c29c4cba0c",
        + 		UserAccountSpecBase: v1alpha1.UserAccountSpecBase{NSTemplateSet: &v1alpha1.NSTemplateSetSpec{}},
        + 	},
          )
```

**Comment by sbryzak** [comment]
@alexeykazakov I can see those failures to match conditions in the logs, but i can't see where they are manifesting as a test error / failure.  I do see the following though:

```
 === RUN   TestProxyFlow/proxymember1/use_proxy_to_create_a_HAS_Application_CR_in_the_user_appstudio_namespace_via_proxy_API_and_use_websocket_to_watch_it_created
    proxy_test.go:276: 
        	Error Trace:	proxy_test.go:276
        	            				proxy_test.go:116
        	Error:      	Received unexpected error:
        	            	websocket: bad handshake
        	Test:       	TestProxyFlow/proxymember1/use_proxy_to_create_a_HAS_Application_CR_in_the_user_appstudio_namespace_via_proxy_API_and_use_websocket_to_watch_it_created
=== CONT  TestProxyFlow ```

Will try to rerun the tests and see if this is repeatable.

**Comment by sbryzak** [comment]
Another e2e test failure, leaving this here for the record:

image

**Comment by sbryzak** [comment]
Another different e2e test failure, once again for the record:

image

**Comment by sbryzak** [comment]
Latest test run fails with yet another AWS error, this time it seems to be route53 related:

image

**Pull request merged**